src/test/java/com/epam/esm/persistence/jdbc/JdbcTemplateGiftDaoImplTest.java

```diff
@@ -0,0 +1,328 @@
+package com.epam.esm.persistence.jdbc;
+
+import com.epam.esm.TestConfig;
+import com.epam.esm.persistence.dao.GiftCertificateDao;
+import com.epam.esm.persistence.jdbc.gift.JdbcTemplateGiftDao;
+import com.epam.esm.persistence.jdbc.gift.JdbcTemplateGiftDaoImpl;
+import com.epam.esm.persistence.util.search.GiftSearchFilter;
+import com.epam.esm.persistence.util.search.QueryOrder;
+import com.epam.esm.util.validation.BeforeGiftValidation;
+import com.epam.esm.web.exception.EntityBadInputException;
+import com.epam.esm.web.exception.EntityNotFoundException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = { TestConfig.class }, loader = AnnotationConfigContextLoader.class)
+@Sql({"classpath:sql/db-schema.sql", "classpath:sql/db-test-data.sql"})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+class JdbcTemplateGiftDaoImplTest {
+
+    private final JdbcTemplateGiftDao jdbcTemplateGiftDao;
+
+    @Autowired
+    public JdbcTemplateGiftDaoImplTest(JdbcTemplate jdbcTemplate,
+                                   BeforeGiftValidation<GiftCertificateDao, Long> giftValidation) {
+        jdbcTemplateGiftDao = new JdbcTemplateGiftDaoImpl(jdbcTemplate, giftValidation);
+    }
+
+    @Test
+    void findAllGiftsWithoutSearchFilterParams() {
+        try {
+            int sizeAfterInit = 4;
+
+            List<GiftCertificateDao> listResult = jdbcTemplateGiftDao.findAllEntities(new GiftSearchFilter());
+
+            assertEquals(sizeAfterInit, listResult.size());
+        } catch (Exception e) {
```

Comment:
Just add exceptions which should not appear to throws clause, try and catch really hurt readability.